packages/blade/src/components/TopNav/__tests__/TabNav.test.stories.tsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import type { StoryFn } from '@storybook/react';
+import { within, userEvent } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+import React from 'react';
+import { TabNav, TabNavItem } from '../TabNav';
+import { Box } from '~components/Box';
+import { HomeIcon } from '~components/Icons';
+
+const sleep = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const TestBasicTabNav: StoryFn<typeof TabNav> = (): React.ReactElement => {
+  return (
+    <TabNav>
+      <TabNavItem icon={HomeIcon} accessibilityLabel="Home" href="/home" />
+      <TabNavItem href="/payroll">Payroll</TabNavItem>
+      <TabNavItem href="/payments">Payments</TabNavItem>
+    </TabNav>
+  );
+};
+
+TestBasicTabNav.play = async ({ canvasElement }) => {
+  const { getByRole } = within(canvasElement);
+  const homeTab = getByRole('link', { name: 'Home' });
+  const payrollTab = getByRole('link', { name: 'Payroll' });
+  const paymentsTab = getByRole('link', { name: 'Payments' });
+  await expect(homeTab).toHaveAttribute('href', '/home');
+  await expect(payrollTab).toHaveAttribute('href', '/payroll');
+  await expect(paymentsTab).toHaveAttribute('href', '/payments');
+};
+
+export const TestOverflow: StoryFn<typeof TabNav> = (): React.ReactElement => {
```

Comment:
no test to check whether clicking item makes it active or not?

Reply:
The selection logic doesn't really work depending on "click" but rather it works same as SideNavLink, isActive gets selected based on page URL. 

The tab itself doesn't do any state management.